diff -pruN csrc.orig/libavcodec/avcodec.h csrc/libavcodec/avcodec.h
--- csrc.orig/libavcodec/avcodec.h	2016-09-19 18:25:37.000000000 +0200
+++ csrc/libavcodec/avcodec.h	2016-09-19 18:27:02.708544372 +0200
@@ -1621,7 +1621,9 @@ enum AVSideDataParamChangeFlags {
  * @}
  */
 
-struct AVCodecInternal;
+struct AVCodecInternal {
+    int dummy;
+};
 
 enum AVFieldOrder {
     AV_FIELD_UNKNOWN,
@@ -3485,7 +3487,9 @@ typedef struct AVProfile {
     const char *name; ///< short name for the profile
 } AVProfile;
 
-typedef struct AVCodecDefault AVCodecDefault;
+typedef struct AVCodecDefault {
+    int dummy;
+} AVCodecDefault;
 
 struct AVSubtitle;
 
diff -pruN csrc.orig/libavfilter/avfilter.h csrc/libavfilter/avfilter.h
--- csrc.orig/libavfilter/avfilter.h	2016-09-19 18:20:56.000000000 +0200
+++ csrc/libavfilter/avfilter.h	2016-09-19 18:27:02.712544436 +0200
@@ -64,8 +64,13 @@ const char *avfilter_license(void);
 
 typedef struct AVFilterContext AVFilterContext;
 typedef struct AVFilterLink    AVFilterLink;
-typedef struct AVFilterPad     AVFilterPad;
-typedef struct AVFilterFormats AVFilterFormats;
+typedef struct AVFilterPad {
+  int dummy;
+} AVFilterPad;
+
+typedef struct AVFilterFormats {
+  int dummy;
+} AVFilterFormats;
 
 /**
  * Get the number of elements in a NULL-terminated array of AVFilterPads (e.g.
@@ -299,7 +304,13 @@ typedef struct AVFilter {
  */
 #define AVFILTER_THREAD_SLICE (1 << 0)
 
-typedef struct AVFilterInternal AVFilterInternal;
+typedef struct AVFilterInternal {
+  int dummy;
+} AVFilterInternal;
+
+typedef struct AVFilterCommand {
+  int dummy;
+} AVFilterCommand;
 
 /** An instance of a filter */
 struct AVFilterContext {
@@ -368,6 +379,10 @@ struct AVFilterContext {
  * which have been negotiated and agreed upon between the filter, such as
  * image dimensions, format, etc.
  */
+typedef AVFilterChannelLayouts {
+  int dummy;
+} AVFilterChannelLayouts;
+
 struct AVFilterLink {
     AVFilterContext *src;       ///< source filter
     AVFilterPad *srcpad;        ///< output pad on the source filter
@@ -738,7 +753,9 @@ int avfilter_insert_filter(AVFilterLink
  */
 const AVClass *avfilter_get_class(void);
 
-typedef struct AVFilterGraphInternal AVFilterGraphInternal;
+typedef struct AVFilterGraphInternal {
+  int dummy;
+} AVFilterGraphInternal;
 
 /**
  * A function pointer passed to the @ref AVFilterGraph.execute callback to be
diff -pruN csrc.orig/libavformat/avformat.h csrc/libavformat/avformat.h
--- csrc.orig/libavformat/avformat.h	2016-09-19 18:23:59.000000000 +0200
+++ csrc/libavformat/avformat.h	2016-09-19 18:27:02.712544436 +0200
@@ -326,6 +326,10 @@ struct AVFormatContext;
 struct AVDeviceInfoList;
 struct AVDeviceCapabilitiesQuery;
 
+typedef struct FFFrac {
+  int dummy;
+} FFFrac;
+
 /**
  * @defgroup metadata_api Public Metadata API
  * @{
@@ -451,7 +455,9 @@ typedef struct AVFrac {
 /*************************************************/
 /* input/output formats */
 
-struct AVCodecTag;
+struct AVCodecTag {
+    int dummy;
+};
 
 /**
  * This structure contains the data a format has to probe a file.
@@ -841,7 +847,9 @@ typedef struct AVIndexEntry {
  */
 #define AV_DISPOSITION_ATTACHED_PIC      0x0400
 
-typedef struct AVStreamInternal AVStreamInternal;
+typedef struct AVStreamInternal {
+  int dummy;
+} AVStreamInternal;
 
 /**
  * To specify text track kind (different from subtitles default).
@@ -1294,7 +1302,9 @@ enum AVDurationEstimationMethod {
     AVFMT_DURATION_FROM_BITRATE ///< Duration estimated from bitrate (less accurate)
 };
 
-typedef struct AVFormatInternal AVFormatInternal;
+typedef struct AVFormatInternal {
+    int dummy;
+} AVFormatInternal;
 
 /**
  * Format I/O context.
diff -pruN csrc.orig/libavutil/buffer.h csrc/libavutil/buffer.h
--- csrc.orig/libavutil/buffer.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/buffer.h	2016-09-19 18:27:02.712544436 +0200
@@ -70,7 +70,9 @@
  * A reference counted buffer type. It is opaque and is meant to be used through
  * references (AVBufferRef).
  */
-typedef struct AVBuffer AVBuffer;
+typedef struct AVBuffer {
+    int dummy;
+} AVBuffer;
 
 /**
  * A reference to a data buffer.
diff -pruN csrc.orig/libavutil/dict.h csrc/libavutil/dict.h
--- csrc.orig/libavutil/dict.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/dict.h	2016-09-19 18:27:02.712544436 +0200
@@ -83,7 +83,9 @@ typedef struct AVDictionaryEntry {
     char *value;
 } AVDictionaryEntry;
 
-typedef struct AVDictionary AVDictionary;
+typedef struct AVDictionary {
+    int dummy;
+} AVDictionary;
 
 /**
  * Get a dictionary entry with matching key.
