diff -pruN csrc.orig/libavcodec/avcodec.h csrc/libavcodec/avcodec.h
--- csrc.orig/libavcodec/avcodec.h	2017-05-07 08:40:46.000000000 +0200
+++ csrc/libavcodec/avcodec.h	2017-05-07 08:41:37.965384394 +0200
@@ -1694,7 +1694,9 @@ enum AVSideDataParamChangeFlags {
  * @}
  */
 
-struct AVCodecInternal;
+struct AVCodecInternal {
+    int dummy;
+};
 
 enum AVFieldOrder {
     AV_FIELD_UNKNOWN,
@@ -3614,7 +3616,9 @@ typedef struct AVProfile {
     const char *name; ///< short name for the profile
 } AVProfile;
 
-typedef struct AVCodecDefault AVCodecDefault;
+typedef struct AVCodecDefault {
+    int dummy;
+} AVCodecDefault;
 
 struct AVSubtitle;
 
@@ -5364,7 +5368,9 @@ int avcodec_encode_subtitle(AVCodecConte
 struct ReSampleContext;
 struct AVResampleContext;
 
-typedef struct ReSampleContext ReSampleContext;
+typedef struct ReSampleContext {
+    int dummy;
+} ReSampleContext;
 
 /**
  *  Initialize audio resampling context.
@@ -5726,7 +5732,9 @@ typedef struct AVBitStreamFilterContext
 } AVBitStreamFilterContext;
 #endif
 
-typedef struct AVBSFInternal AVBSFInternal;
+typedef struct AVBSFInternal {
+    int dummy;
+} AVBSFInternal;
 
 /**
  * The bitstream filter state.
@@ -5998,7 +6006,9 @@ const AVClass *av_bsf_get_class(void);
  * Structure for chain/list of bitstream filters.
  * Empty list can be allocated by av_bsf_list_alloc().
  */
-typedef struct AVBSFList AVBSFList;
+typedef struct AVBSFList {
+  int dummy;
+} AVBSFList;
 
 /**
  * Allocate empty list of bitstream filters.
diff -pruN csrc.orig/libavcodec/avfft.h csrc/libavcodec/avfft.h
--- csrc.orig/libavcodec/avfft.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavcodec/avfft.h	2017-05-07 08:41:37.969384455 +0200
@@ -38,7 +38,9 @@ typedef struct FFTComplex {
     FFTSample re, im;
 } FFTComplex;
 
-typedef struct FFTContext FFTContext;
+typedef struct FFTContext {
+    int dummy;
+} FFTContext;
 
 /**
  * Set up a complex FFT.
@@ -75,7 +77,9 @@ enum RDFTransformType {
     DFT_C2R,
 };
 
-typedef struct RDFTContext RDFTContext;
+typedef struct RDFTContext {
+    int dummy;
+} RDFTContext;
 
 /**
  * Set up a real FFT.
@@ -88,7 +92,9 @@ void av_rdft_end(RDFTContext *s);
 
 /* Discrete Cosine Transform */
 
-typedef struct DCTContext DCTContext;
+typedef struct DCTContext {
+  int dummy;
+} DCTContext;
 
 enum DCTTransformType {
     DCT_II = 0,
diff -pruN csrc.orig/libavcodec/mediacodec.h csrc/libavcodec/mediacodec.h
--- csrc.orig/libavcodec/mediacodec.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavcodec/mediacodec.h	2017-05-07 08:41:37.969384455 +0200
@@ -70,7 +70,9 @@ void av_mediacodec_default_free(AVCodecC
 /**
  * Opaque structure representing a MediaCodec buffer to render.
  */
-typedef struct MediaCodecBuffer AVMediaCodecBuffer;
+typedef struct MediaCodecBuffer {
+  int dummy;
+} AVMediaCodecBuffer;
 
 /**
  * Release a MediaCodec buffer and render it to the surface that is associated
diff -pruN csrc.orig/libavcodec/vorbis_parser.h csrc/libavcodec/vorbis_parser.h
--- csrc.orig/libavcodec/vorbis_parser.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavcodec/vorbis_parser.h	2017-05-07 08:41:37.969384455 +0200
@@ -28,7 +28,9 @@
 
 #include <stdint.h>
 
-typedef struct AVVorbisParseContext AVVorbisParseContext;
+typedef struct AVVorbisParseContext {
+    int dummy;
+} AVVorbisParseContext;
 
 /**
  * Allocate and initialize the Vorbis parser using headers in the extradata.
diff -pruN csrc.orig/libavfilter/avfilter.h csrc/libavfilter/avfilter.h
--- csrc.orig/libavfilter/avfilter.h	2017-05-07 08:37:59.000000000 +0200
+++ csrc/libavfilter/avfilter.h	2017-05-07 08:41:37.969384455 +0200
@@ -64,10 +64,19 @@ const char *avfilter_configuration(void)
  */
 const char *avfilter_license(void);
 
-typedef struct AVFilterContext AVFilterContext;
-typedef struct AVFilterLink    AVFilterLink;
-typedef struct AVFilterPad     AVFilterPad;
-typedef struct AVFilterFormats AVFilterFormats;
+typedef struct AVFilterContext {
+    int dummy;
+} AVFilterContext;
+typedef struct AVFilterLink {
+    int dummy;
+} AVFilterLink;
+typedef struct AVFilterPad {
+  int dummy;
+} AVFilterPad;
+
+typedef struct AVFilterFormats {
+  int dummy;
+} AVFilterFormats;
 
 /**
  * Get the number of elements in a NULL-terminated array of AVFilterPads (e.g.
@@ -317,7 +326,13 @@ typedef struct AVFilter {
  */
 #define AVFILTER_THREAD_SLICE (1 << 0)
 
-typedef struct AVFilterInternal AVFilterInternal;
+typedef struct AVFilterInternal {
+  int dummy;
+} AVFilterInternal;
+
+typedef struct AVFilterCommand {
+  int dummy;
+} AVFilterCommand;
 
 /** An instance of a filter */
 struct AVFilterContext {
@@ -405,6 +420,10 @@ struct AVFilterContext {
  * In the future, access to the header may be reserved for filters
  * implementation.
  */
+typedef AVFilterChannelLayouts {
+  int dummy;
+} AVFilterChannelLayouts;
+
 struct AVFilterLink {
     AVFilterContext *src;       ///< source filter
     AVFilterPad *srcpad;        ///< output pad on the source filter
@@ -801,7 +820,9 @@ int avfilter_insert_filter(AVFilterLink
  */
 const AVClass *avfilter_get_class(void);
 
-typedef struct AVFilterGraphInternal AVFilterGraphInternal;
+typedef struct AVFilterGraphInternal {
+  int dummy;
+} AVFilterGraphInternal;
 
 /**
  * A function pointer passed to the @ref AVFilterGraph.execute callback to be
diff -pruN csrc.orig/libavformat/avformat.h csrc/libavformat/avformat.h
--- csrc.orig/libavformat/avformat.h	2017-05-07 08:36:29.000000000 +0200
+++ csrc/libavformat/avformat.h	2017-05-07 08:41:37.973384516 +0200
@@ -328,6 +328,10 @@ struct AVFormatContext;
 struct AVDeviceInfoList;
 struct AVDeviceCapabilitiesQuery;
 
+typedef struct FFFrac {
+  int dummy;
+} FFFrac;
+
 /**
  * @defgroup metadata_api Public Metadata API
  * @{
@@ -453,7 +457,9 @@ typedef struct AVFrac {
 /*************************************************/
 /* input/output formats */
 
-struct AVCodecTag;
+struct AVCodecTag {
+    int dummy;
+};
 
 /**
  * This structure contains the data a format has to probe a file.
@@ -854,7 +860,9 @@ typedef struct AVIndexEntry {
  */
 #define AV_DISPOSITION_TIMED_THUMBNAILS  0x0800
 
-typedef struct AVStreamInternal AVStreamInternal;
+typedef struct AVStreamInternal {
+  int dummy;
+} AVStreamInternal;
 
 /**
  * To specify text track kind (different from subtitles default).
@@ -1318,7 +1326,9 @@ enum AVDurationEstimationMethod {
     AVFMT_DURATION_FROM_BITRATE ///< Duration estimated from bitrate (less accurate)
 };
 
-typedef struct AVFormatInternal AVFormatInternal;
+typedef struct AVFormatInternal {
+    int dummy;
+} AVFormatInternal;
 
 /**
  * Format I/O context.
diff -pruN csrc.orig/libavresample/avresample.h csrc/libavresample/avresample.h
--- csrc.orig/libavresample/avresample.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavresample/avresample.h	2017-05-07 08:41:37.973384516 +0200
@@ -103,7 +103,9 @@
 
 #define AVRESAMPLE_MAX_CHANNELS 32
 
-typedef struct AVAudioResampleContext AVAudioResampleContext;
+typedef struct AVAudioResampleContext {
+    int dummy;
+} AVAudioResampleContext;
 
 /** Mixing Coefficient Types */
 enum AVMixCoeffType {
diff -pruN csrc.orig/libavutil/audio_fifo.h csrc/libavutil/audio_fifo.h
--- csrc.orig/libavutil/audio_fifo.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/audio_fifo.h	2017-05-07 08:41:37.973384516 +0200
@@ -46,7 +46,9 @@
  * - Supports multiple channels with either planar or packed sample format.
  * - Automatic reallocation when writing to a full buffer.
  */
-typedef struct AVAudioFifo AVAudioFifo;
+typedef struct AVAudioFifo {
+    int dummy;
+} AVAudioFifo;
 
 /**
  * Free an AVAudioFifo.
diff -pruN csrc.orig/libavutil/buffer.h csrc/libavutil/buffer.h
--- csrc.orig/libavutil/buffer.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/buffer.h	2017-05-07 08:41:37.973384516 +0200
@@ -70,7 +70,9 @@
  * A reference counted buffer type. It is opaque and is meant to be used through
  * references (AVBufferRef).
  */
-typedef struct AVBuffer AVBuffer;
+typedef struct AVBuffer {
+    int dummy;
+} AVBuffer;
 
 /**
  * A reference to a data buffer.
@@ -235,7 +237,9 @@ int av_buffer_realloc(AVBufferRef **buf,
  * directly. It is allocated with av_buffer_pool_init() and freed with
  * av_buffer_pool_uninit().
  */
-typedef struct AVBufferPool AVBufferPool;
+typedef struct AVBufferPool {
+    int dummy;
+} AVBufferPool;
 
 /**
  * Allocate and initialize a buffer pool.
diff -pruN csrc.orig/libavutil/dict.h csrc/libavutil/dict.h
--- csrc.orig/libavutil/dict.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/dict.h	2017-05-07 08:41:37.973384516 +0200
@@ -83,7 +83,9 @@ typedef struct AVDictionaryEntry {
     char *value;
 } AVDictionaryEntry;
 
-typedef struct AVDictionary AVDictionary;
+typedef struct AVDictionary {
+    int dummy;
+} AVDictionary;
 
 /**
  * Get a dictionary entry with matching key.
diff -pruN csrc.orig/libavutil/eval.h csrc/libavutil/eval.h
--- csrc.orig/libavutil/eval.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/eval.h	2017-05-07 08:41:37.973384516 +0200
@@ -28,7 +28,9 @@
 
 #include "avutil.h"
 
-typedef struct AVExpr AVExpr;
+typedef struct AVExpr {
+    int dummy;
+} AVExpr;
 
 /**
  * Parse and evaluate an expression.
diff -pruN csrc.orig/libavutil/hmac.h csrc/libavutil/hmac.h
--- csrc.orig/libavutil/hmac.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/hmac.h	2017-05-07 08:41:37.973384516 +0200
@@ -39,7 +39,9 @@ enum AVHMACType {
     AV_HMAC_SHA512,
 };
 
-typedef struct AVHMAC AVHMAC;
+typedef struct AVHMAC {
+    int dummy;
+} AVHMAC;
 
 /**
  * Allocate an AVHMAC context.
diff -pruN csrc.orig/libavutil/hwcontext.h csrc/libavutil/hwcontext.h
--- csrc.orig/libavutil/hwcontext.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/hwcontext.h	2017-05-07 08:41:37.973384516 +0200
@@ -32,7 +32,9 @@ enum AVHWDeviceType {
     AV_HWDEVICE_TYPE_QSV,
 };
 
-typedef struct AVHWDeviceInternal AVHWDeviceInternal;
+typedef struct AVHWDeviceInternal {
+    int dummy;
+} AVHWDeviceInternal;
 
 /**
  * This struct aggregates all the (hardware/vendor-specific) "high-level" state,
@@ -102,7 +104,9 @@ typedef struct AVHWDeviceContext {
     void *user_opaque;
 } AVHWDeviceContext;
 
-typedef struct AVHWFramesInternal AVHWFramesInternal;
+typedef struct AVHWFramesInternal {
+    int dummy;
+} AVHWFramesInternal;
 
 /**
  * This struct describes a set or pool of "hardware" frames (i.e. those with
diff -pruN csrc.orig/libavutil/threadmessage.h csrc/libavutil/threadmessage.h
--- csrc.orig/libavutil/threadmessage.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libavutil/threadmessage.h	2017-05-07 08:41:37.973384516 +0200
@@ -19,7 +19,9 @@
 #ifndef AVUTIL_THREADMESSAGE_H
 #define AVUTIL_THREADMESSAGE_H
 
-typedef struct AVThreadMessageQueue AVThreadMessageQueue;
+typedef struct AVThreadMessageQueue {
+    int dummy;
+} AVThreadMessageQueue;
 
 typedef enum AVThreadMessageFlags {
 
diff -pruN csrc.orig/libswresample/swresample.h csrc/libswresample/swresample.h
--- csrc.orig/libswresample/swresample.h	2017-04-28 08:36:02.000000000 +0200
+++ csrc/libswresample/swresample.h	2017-05-07 08:41:37.977384577 +0200
@@ -183,7 +183,9 @@ enum SwrFilterType {
  * the @ref avoptions API and cannot directly set values to members of the
  * structure.
  */
-typedef struct SwrContext SwrContext;
+typedef struct SwrContext {
+    int dummy;
+} SwrContext;
 
 /**
  * Get the AVClass for SwrContext. It can be used in combination with
