diff -pruN csrc.orig/libavcodec/version.h csrc/libavcodec/version.h
--- csrc.orig/libavcodec/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavcodec/version.h	2016-12-23 17:17:56.811586705 +0100
@@ -39,7 +39,7 @@
                                            LIBAVCODEC_VERSION_MICRO)
 #define LIBAVCODEC_BUILD        LIBAVCODEC_VERSION_INT
 
-#define LIBAVCODEC_IDENT        "Lavc" AV_STRINGIFY(LIBAVCODEC_VERSION)
+#define LIBAVCODEC_IDENT        "Lavc57.64.100"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
diff -pruN csrc.orig/libavdevice/version.h csrc/libavdevice/version.h
--- csrc.orig/libavdevice/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavdevice/version.h	2016-12-23 17:18:09.212135725 +0100
@@ -39,7 +39,7 @@
                                            LIBAVDEVICE_VERSION_MICRO)
 #define LIBAVDEVICE_BUILD       LIBAVDEVICE_VERSION_INT
 
-#define LIBAVDEVICE_IDENT       "Lavd" AV_STRINGIFY(LIBAVDEVICE_VERSION)
+#define LIBAVDEVICE_IDENT       "Lavd57.1.100"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
diff -pruN csrc.orig/libavfilter/version.h csrc/libavfilter/version.h
--- csrc.orig/libavfilter/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavfilter/version.h	2016-12-23 17:18:24.184798622 +0100
@@ -41,7 +41,7 @@
                                            LIBAVFILTER_VERSION_MICRO)
 #define LIBAVFILTER_BUILD       LIBAVFILTER_VERSION_INT
 
-#define LIBAVFILTER_IDENT       "Lavfi" AV_STRINGIFY(LIBAVFILTER_VERSION)
+#define LIBAVFILTER_IDENT       "Lavfi6.65.100"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
diff -pruN csrc.orig/libavformat/version.h csrc/libavformat/version.h
--- csrc.orig/libavformat/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavformat/version.h	2016-12-23 17:18:37.189374385 +0100
@@ -43,7 +43,7 @@
                                            LIBAVFORMAT_VERSION_MICRO)
 #define LIBAVFORMAT_BUILD       LIBAVFORMAT_VERSION_INT
 
-#define LIBAVFORMAT_IDENT       "Lavf" AV_STRINGIFY(LIBAVFORMAT_VERSION)
+#define LIBAVFORMAT_IDENT       "Lavf57.56.100"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
diff -pruN csrc.orig/libavresample/version.h csrc/libavresample/version.h
--- csrc.orig/libavresample/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavresample/version.h	2016-12-23 17:18:45.389737446 +0100
@@ -39,7 +39,7 @@
                                                   LIBAVRESAMPLE_VERSION_MICRO)
 #define LIBAVRESAMPLE_BUILD        LIBAVRESAMPLE_VERSION_INT
 
-#define LIBAVRESAMPLE_IDENT        "Lavr" AV_STRINGIFY(LIBAVRESAMPLE_VERSION)
+#define LIBAVRESAMPLE_IDENT        "Lavr3.1.0"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
diff -pruN csrc.orig/libavutil/version.h csrc/libavutil/version.h
--- csrc.orig/libavutil/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libavutil/version.h	2016-12-23 17:19:08.450758446 +0100
@@ -90,7 +90,7 @@
                                            LIBAVUTIL_VERSION_MICRO)
 #define LIBAVUTIL_BUILD         LIBAVUTIL_VERSION_INT
 
-#define LIBAVUTIL_IDENT         "Lavu" AV_STRINGIFY(LIBAVUTIL_VERSION)
+#define LIBAVUTIL_IDENT         "Lavu55.34.100"
 
 /**
  * @defgroup lavu_depr_guards Deprecation Guards
diff -pruN csrc.orig/libpostproc/version.h csrc/libpostproc/version.h
--- csrc.orig/libpostproc/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libpostproc/version.h	2016-12-23 17:19:21.079317561 +0100
@@ -40,7 +40,7 @@
                                            LIBPOSTPROC_VERSION_MICRO)
 #define LIBPOSTPROC_BUILD       LIBPOSTPROC_VERSION_INT
 
-#define LIBPOSTPROC_IDENT       "postproc" AV_STRINGIFY(LIBPOSTPROC_VERSION)
+#define LIBPOSTPROC_IDENT       "postproc54.1.100"
 
 #ifndef FF_API_QP_TYPE
 #define FF_API_QP_TYPE     (LIBPOSTPROC_VERSION_MAJOR < 55)
diff -pruN csrc.orig/libswresample/version.h csrc/libswresample/version.h
--- csrc.orig/libswresample/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libswresample/version.h	2016-12-23 17:19:32.043803000 +0100
@@ -40,6 +40,6 @@
                                               LIBSWRESAMPLE_VERSION_MICRO)
 #define LIBSWRESAMPLE_BUILD        LIBSWRESAMPLE_VERSION_INT
 
-#define LIBSWRESAMPLE_IDENT        "SwR" AV_STRINGIFY(LIBSWRESAMPLE_VERSION)
+#define LIBSWRESAMPLE_IDENT        "SwR2.3.100"
 
 #endif /* SWRESAMPLE_VERSION_H */
diff -pruN csrc.orig/libswscale/version.h csrc/libswscale/version.h
--- csrc.orig/libswscale/version.h	2016-12-15 13:19:26.000000000 +0100
+++ csrc/libswscale/version.h	2016-12-23 17:19:44.064335196 +0100
@@ -38,7 +38,7 @@
                                            LIBSWSCALE_VERSION_MICRO)
 #define LIBSWSCALE_BUILD        LIBSWSCALE_VERSION_INT
 
-#define LIBSWSCALE_IDENT        "SwS" AV_STRINGIFY(LIBSWSCALE_VERSION)
+#define LIBSWSCALE_IDENT        "SwS4.2.100"
 
 /**
  * FF_API_* defines may be placed below to indicate public API that will be
