diff -pruN csrc.orig/libavutil/avstring.h csrc/libavutil/avstring.h
--- csrc.orig/libavutil/avstring.h	2021-06-13 22:30:13.421065529 +0200
+++ csrc/libavutil/avstring.h	2021-06-13 22:44:48.865443337 +0200
@@ -222,15 +222,6 @@ static inline av_const int av_isgraph(in
 }
 
 /**
- * Locale-independent conversion of ASCII isspace.
- */
-static inline av_const int av_isspace(int c)
-{
-    return c == ' ' || c == '\f' || c == '\n' || c == '\r' || c == '\t' ||
-           c == '\v';
-}
-
-/**
  * Locale-independent conversion of ASCII characters to uppercase.
  */
 static inline av_const int av_toupper(int c)
