diff -pruN csrc.orig/libavcodec/avcodec.h csrc/libavcodec/avcodec.h
--- csrc.orig/libavcodec/avcodec.h	2016-05-02 14:52:08.000000000 +0200
+++ csrc/libavcodec/avcodec.h	2016-05-03 17:25:53.253843286 +0200
@@ -1799,7 +1799,7 @@ typedef struct AVCodecContext {
      * - encoding: unused
      * - decoding: Set by user, if not set the native format will be chosen.
      */
-    enum AVPixelFormat (*get_format)(struct AVCodecContext *s, const enum AVPixelFormat * fmt);
+    long get_format;
 
     /**
      * maximum number of B-frames between non-B-frames
@@ -5335,7 +5335,7 @@ enum AVLockOp {
  *           implement your lock manager). If cb is set to NULL the
  *           lockmgr will be unregistered.
  */
-int av_lockmgr_register(int (*cb)(void **mutex, enum AVLockOp op));
+//int av_lockmgr_register(int (*cb)(void **mutex, enum AVLockOp op));
 
 /**
  * Get the type of the given codec.
