diff -pruN csrc.orig/libavcodec/avcodec.h csrc/libavcodec/avcodec.h
--- csrc.orig/libavcodec/avcodec.h	2016-09-19 18:27:02.000000000 +0200
+++ csrc/libavcodec/avcodec.h	2016-09-20 16:04:57.560387227 +0200
@@ -5225,7 +5225,9 @@ int avcodec_encode_subtitle(AVCodecConte
 struct ReSampleContext;
 struct AVResampleContext;
 
-typedef struct ReSampleContext ReSampleContext;
+typedef struct ReSampleContext {
+    int dummy;
+} ReSampleContext;
 
 /**
  *  Initialize audio resampling context.
@@ -5590,7 +5592,9 @@ typedef struct AVBitStreamFilterContext
 } AVBitStreamFilterContext;
 #endif
 
-typedef struct AVBSFInternal AVBSFInternal;
+typedef struct AVBSFInternal {
+    int dummy;
+} AVBSFInternal;
 
 /**
  * The bitstream filter state.
diff -pruN csrc.orig/libavcodec/avfft.h csrc/libavcodec/avfft.h
--- csrc.orig/libavcodec/avfft.h	2016-08-28 15:49:56.000000000 +0200
+++ csrc/libavcodec/avfft.h	2016-09-20 16:05:38.312997926 +0200
@@ -38,7 +38,9 @@ typedef struct FFTComplex {
     FFTSample re, im;
 } FFTComplex;
 
-typedef struct FFTContext FFTContext;
+typedef struct FFTContext {
+    int dummy;
+} FFTContext;
 
 /**
  * Set up a complex FFT.
@@ -75,7 +77,9 @@ enum RDFTransformType {
     DFT_C2R,
 };
 
-typedef struct RDFTContext RDFTContext;
+typedef struct RDFTContext {
+    int dummy;
+} RDFTContext;
 
 /**
  * Set up a real FFT.
diff -pruN csrc.orig/libavcodec/vorbis_parser.h csrc/libavcodec/vorbis_parser.h
--- csrc.orig/libavcodec/vorbis_parser.h	2016-08-28 15:49:56.000000000 +0200
+++ csrc/libavcodec/vorbis_parser.h	2016-09-20 16:05:57.213281776 +0200
@@ -28,7 +28,9 @@
 
 #include <stdint.h>
 
-typedef struct AVVorbisParseContext AVVorbisParseContext;
+typedef struct AVVorbisParseContext {
+    int dummy;
+} AVVorbisParseContext;
 
 /**
  * Allocate and initialize the Vorbis parser using headers in the extradata.
diff -pruN csrc.orig/libavfilter/avfilter.h csrc/libavfilter/avfilter.h
--- csrc.orig/libavfilter/avfilter.h	2016-09-19 18:27:02.000000000 +0200
+++ csrc/libavfilter/avfilter.h	2016-09-20 16:06:24.237688245 +0200
@@ -62,8 +62,12 @@ const char *avfilter_configuration(void)
  */
 const char *avfilter_license(void);
 
-typedef struct AVFilterContext AVFilterContext;
-typedef struct AVFilterLink    AVFilterLink;
+typedef struct AVFilterContext {
+    int dummy;
+} AVFilterContext;
+typedef struct AVFilterLink {
+    int dummy;
+} AVFilterLink;
 typedef struct AVFilterPad {
   int dummy;
 } AVFilterPad;
diff -pruN csrc.orig/libavresample/avresample.h csrc/libavresample/avresample.h
--- csrc.orig/libavresample/avresample.h	2016-08-28 15:49:56.000000000 +0200
+++ csrc/libavresample/avresample.h	2016-09-20 16:06:57.734192958 +0200
@@ -103,7 +103,9 @@
 
 #define AVRESAMPLE_MAX_CHANNELS 32
 
-typedef struct AVAudioResampleContext AVAudioResampleContext;
+typedef struct AVAudioResampleContext {
+    int dummy;
+} AVAudioResampleContext;
 
 /** Mixing Coefficient Types */
 enum AVMixCoeffType {
diff -pruN csrc.orig/libavutil/audio_fifo.h csrc/libavutil/audio_fifo.h
--- csrc.orig/libavutil/audio_fifo.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/audio_fifo.h	2016-09-20 16:07:08.878361074 +0200
@@ -46,7 +46,9 @@
  * - Supports multiple channels with either planar or packed sample format.
  * - Automatic reallocation when writing to a full buffer.
  */
-typedef struct AVAudioFifo AVAudioFifo;
+typedef struct AVAudioFifo {
+    int dummy;
+} AVAudioFifo;
 
 /**
  * Free an AVAudioFifo.
diff -pruN csrc.orig/libavutil/buffer.h csrc/libavutil/buffer.h
--- csrc.orig/libavutil/buffer.h	2016-09-19 18:27:02.000000000 +0200
+++ csrc/libavutil/buffer.h	2016-09-20 16:07:20.390534839 +0200
@@ -237,7 +237,9 @@ int av_buffer_realloc(AVBufferRef **buf,
  * directly. It is allocated with av_buffer_pool_init() and freed with
  * av_buffer_pool_uninit().
  */
-typedef struct AVBufferPool AVBufferPool;
+typedef struct AVBufferPool {
+    int dummy;
+} AVBufferPool;
 
 /**
  * Allocate and initialize a buffer pool.
diff -pruN csrc.orig/libavutil/eval.h csrc/libavutil/eval.h
--- csrc.orig/libavutil/eval.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/eval.h	2016-09-20 16:07:35.902769131 +0200
@@ -28,7 +28,9 @@
 
 #include "avutil.h"
 
-typedef struct AVExpr AVExpr;
+typedef struct AVExpr {
+    int dummy;
+} AVExpr;
 
 /**
  * Parse and evaluate an expression.
diff -pruN csrc.orig/libavutil/hmac.h csrc/libavutil/hmac.h
--- csrc.orig/libavutil/hmac.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/hmac.h	2016-09-20 16:07:47.782948676 +0200
@@ -39,7 +39,9 @@ enum AVHMACType {
     AV_HMAC_SHA512,
 };
 
-typedef struct AVHMAC AVHMAC;
+typedef struct AVHMAC {
+    int dummy;
+} AVHMAC;
 
 /**
  * Allocate an AVHMAC context.
diff -pruN csrc.orig/libavutil/hwcontext.h csrc/libavutil/hwcontext.h
--- csrc.orig/libavutil/hwcontext.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/hwcontext.h	2016-09-20 16:08:24.167499100 +0200
@@ -31,7 +31,9 @@ enum AVHWDeviceType {
     AV_HWDEVICE_TYPE_DXVA2,
 };
 
-typedef struct AVHWDeviceInternal AVHWDeviceInternal;
+typedef struct AVHWDeviceInternal {
+    int dummy;
+} AVHWDeviceInternal;
 
 /**
  * This struct aggregates all the (hardware/vendor-specific) "high-level" state,
@@ -101,7 +103,9 @@ typedef struct AVHWDeviceContext {
     void *user_opaque;
 } AVHWDeviceContext;
 
-typedef struct AVHWFramesInternal AVHWFramesInternal;
+typedef struct AVHWFramesInternal {
+    int dummy;
+} AVHWFramesInternal;
 
 /**
  * This struct describes a set or pool of "hardware" frames (i.e. those with
diff -pruN csrc.orig/libavutil/threadmessage.h csrc/libavutil/threadmessage.h
--- csrc.orig/libavutil/threadmessage.h	2016-08-28 15:49:57.000000000 +0200
+++ csrc/libavutil/threadmessage.h	2016-09-20 16:07:59.559126738 +0200
@@ -19,7 +19,9 @@
 #ifndef AVUTIL_THREADMESSAGE_H
 #define AVUTIL_THREADMESSAGE_H
 
-typedef struct AVThreadMessageQueue AVThreadMessageQueue;
+typedef struct AVThreadMessageQueue {
+    int dummy;
+} AVThreadMessageQueue;
 
 typedef enum AVThreadMessageFlags {
 
diff -pruN csrc.orig/libswresample/swresample.h csrc/libswresample/swresample.h
--- csrc.orig/libswresample/swresample.h	2016-08-28 15:49:56.000000000 +0200
+++ csrc/libswresample/swresample.h	2016-09-20 16:08:54.327955931 +0200
@@ -183,7 +183,9 @@ enum SwrFilterType {
  * the @ref avoptions API and cannot directly set values to members of the
  * structure.
  */
-typedef struct SwrContext SwrContext;
+typedef struct SwrContext {
+    int dummy;
+} SwrContext;
 
 /**
  * Get the AVClass for SwrContext. It can be used in combination with
