diff -up csrc/libavutil/avstring.h.orig csrc/libavutil/avstring.h
--- csrc/libavutil/avstring.h.orig	2018-02-19 10:29:18.000000000 +0100
+++ csrc/libavutil/avstring.h	2018-04-22 09:44:02.136022586 +0200
@@ -217,15 +217,6 @@ static inline av_const int av_isgraph(in
 }
 
 /**
- * Locale-independent conversion of ASCII isspace.
- */
-static inline av_const int av_isspace(int c)
-{
-    return c == ' ' || c == '\f' || c == '\n' || c == '\r' || c == '\t' ||
-           c == '\v';
-}
-
-/**
  * Locale-independent conversion of ASCII characters to uppercase.
  */
 static inline av_const int av_toupper(int c)
