diff -pruN csrc.orig/libavcodec/avcodec.h csrc/libavcodec/avcodec.h
--- csrc.orig/libavcodec/avcodec.h	2015-09-27 12:09:29.000000000 +0200
+++ csrc/libavcodec/avcodec.h	2015-09-27 12:11:35.388745614 +0200
@@ -1003,7 +1003,9 @@ enum AVSideDataParamChangeFlags {
  * @}
  */
 
-struct AVCodecInternal;
+struct AVCodecInternal {
+    int dummy;
+};
 
 enum AVFieldOrder {
     AV_FIELD_UNKNOWN,
@@ -2967,7 +2969,9 @@ typedef struct AVProfile {
     const char *name; ///< short name for the profile
 } AVProfile;
 
-typedef struct AVCodecDefault AVCodecDefault;
+typedef struct AVCodecDefault {
+    int dummy;
+} AVCodecDefault;
 
 struct AVSubtitle;
 
diff -pruN csrc.orig/libavformat/avformat.h csrc/libavformat/avformat.h
--- csrc.orig/libavformat/avformat.h	2015-09-27 12:09:29.000000000 +0200
+++ csrc/libavformat/avformat.h	2015-09-27 12:11:35.390745609 +0200
@@ -440,7 +440,9 @@ typedef struct AVFrac {
 /*************************************************/
 /* input/output formats */
 
-struct AVCodecTag;
+struct AVCodecTag {
+    int dummy;
+};
 
 /**
  * This structure contains the data a format has to probe a file.
@@ -1250,7 +1252,9 @@ enum AVDurationEstimationMethod {
     AVFMT_DURATION_FROM_BITRATE ///< Duration estimated from bitrate (less accurate)
 };
 
-typedef struct AVFormatInternal AVFormatInternal;
+typedef struct AVFormatInternal {
+    int dummy;
+} AVFormatInternal;
 
 /**
  * Format I/O context.
diff -pruN csrc.orig/libavutil/buffer.h csrc/libavutil/buffer.h
--- csrc.orig/libavutil/buffer.h	2015-09-19 11:18:47.000000000 +0200
+++ csrc/libavutil/buffer.h	2015-09-27 12:11:35.391745607 +0200
@@ -70,7 +70,9 @@
  * A reference counted buffer type. It is opaque and is meant to be used through
  * references (AVBufferRef).
  */
-typedef struct AVBuffer AVBuffer;
+typedef struct AVBuffer {
+    int dummy;
+} AVBuffer;
 
 /**
  * A reference to a data buffer.
diff -pruN csrc.orig/libavutil/dict.h csrc/libavutil/dict.h
--- csrc.orig/libavutil/dict.h	2015-09-19 11:18:47.000000000 +0200
+++ csrc/libavutil/dict.h	2015-09-27 12:11:35.392745604 +0200
@@ -84,7 +84,9 @@ typedef struct AVDictionaryEntry {
     char *value;
 } AVDictionaryEntry;
 
-typedef struct AVDictionary AVDictionary;
+typedef struct AVDictionary {
+    int dummy;
+} AVDictionary;
 
 /**
  * Get a dictionary entry with matching key.
