diff -up csrc/libavutil/bprint.h.orig csrc/libavutil/bprint.h
--- csrc/libavutil/bprint.h.orig	2018-02-19 10:29:18.000000000 +0100
+++ csrc/libavutil/bprint.h	2018-04-22 15:10:32.768651269 +0200
@@ -79,13 +79,14 @@ typedef struct name { \
  * such as the current paragraph.
  */
 
-FF_PAD_STRUCTURE(AVBPrint, 1024,
+typedef struct AVBPrint {
     char *str;         /**< string so far */
     unsigned len;      /**< length so far */
     unsigned size;     /**< allocated memory */
     unsigned size_max; /**< maximum allocated memory */
     char reserved_internal_buffer[1];
-)
+    char reserved_padding[1024 - 17];
+};
 
 /**
  * Convenience macros for special values for av_bprint_init() size_max
